web/modules/custom/dpl_event/dpl_event.module

```diff
@@ -294,19 +294,52 @@ function dpl_event_preprocess_eventinstance__list_teaser(array &$variables): voi
     }
   }
 
-  // Extract start_date, end_date and datetime attributes.
+  // Extract start_time, end_time and datetime attributes.
   if (!$eventInstance->get('date')->isEmpty()) {
     $date_field = $eventInstance->get('date')->first();
 
-    $start_date = $date_field->start_date;
-    $end_date = $date_field->end_date;
+    $start_time = $date_field->start_date;
+    $end_time = $date_field->end_date;
 
-    if ($start_date instanceof DrupalDateTime) {
-      $variables['start_date'] = $date_formatter->format($start_date->getTimestamp(), 'custom', 'H:i');
-      $variables['datetime_attribute'] = $date_formatter->format($start_date->getTimestamp(), DATE_ATOM);
+    if ($start_time instanceof DrupalDateTime) {
+      $variables['start_time'] = $date_formatter->format($start_time->getTimestamp(), 'custom', 'H:i');
+      $variables['datetime_attribute'] = $date_formatter->format($start_time->getTimestamp(), DATE_ATOM);
     }
-    if ($end_date instanceof DrupalDateTime) {
-      $variables['end_date'] = $date_formatter->format($end_date->getTimestamp(), 'custom', 'H:i');
+    if ($end_time instanceof DrupalDateTime) {
+      $variables['end_time'] = $date_formatter->format($end_time->getTimestamp(), 'custom', 'H:i');
+    }
+  }
+}
+
+/**
+ * Implements hook_preprocess_eventinstance__stacked_event().
+ *
+ * Preprocesses variables for the 'list_teaser' view mode of eventinstance.
+ * - Generating the URL for the full view of the event instance.
+ * - Processing start and end dates from the 'date' field and formatting them
+ *   in the site's timezone.
+ */
+function dpl_event_preprocess_eventinstance__stacked_event(array &$variables): void {
+  $eventInstance = $variables['eventinstance'];
+  /** @var \Drupal\Core\Datetime\DateFormatterInterface $date_formatter */
+  $date_formatter = \Drupal::service('date.formatter');
+
+  // Extract the event url.
+  $variables['eventinstance_url'] = Url::fromRoute('entity.eventinstance.canonical', ['eventinstance' => $eventInstance->id()])->toString();
+
+  // Extract start_time, end_time and datetime attributes.
+  if (!$eventInstance->get('date')->isEmpty()) {
+    $date_field = $eventInstance->get('date')->first();
+
+    $start_time = $date_field->start_date;
+    $end_time = $date_field->end_date;
+
+    if ($start_time instanceof DrupalDateTime) {
+      $variables['start_time'] = $date_formatter->format($start_time->getTimestamp(), 'custom', 'H:i');
+      $variables['datetime_attribute'] = $date_formatter->format($start_time->getTimestamp(), DATE_ATOM);
+    }
+    if ($end_time instanceof DrupalDateTime) {
+      $variables['end_time'] = $date_formatter->format($end_time->getTimestamp(), 'custom', 'H:i');
```

Comment:
Please check if you can move extraction of url and dates to a new preprocess function which is used across view modes: `dpl_event_preprocess_eventinstance()`.

The conent seems to be partly duplicated from what is going on in `dpl_event_preprocess_eventinstance__list_teaser()`.